### PR TITLE
Move to granular opt-in for interceptors preview

### PR DESCRIPTION
**Pull request by captainsafia**

### CODE REVIEW

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToUseAnalyzers.cs [hunk 1]
```diff
@@ -50,7 +50,7 @@ public void It_resolves_configbindinggenerator_correctly(string testAssetName, b
                 });
 
             VerifyConfigBindingGeneratorIsUsed(asset, isEnabled);
-            VerifyInterceptorsFeatureEnabled(asset, isEnabled);
+            VerifyInterceptorsFeatureEnabled(asset, isEnabled, "Microsoft.Extensions.Configuration.Binder.SourceGeneration");
```
Comment (layomia) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToUseAnalyzers.cs:53]: nit: these `Verify*` method names read like they (always) assert that the generator(s) are used/enabled. Consider using names that indicate that the expected behavior depends on whether the generator is enabled (e.g VerifyInterceptorsFeatureSettingIsCorrect).

File: src/WebSdk/ProjectSystem/Targets/Microsoft.NET.Sdk.Web.ProjectSystem.targets [hunk 2]
```diff
@@ -59,9 +59,13 @@ Copyright (c) .NET Foundation. All rights reserved.
     <EnableConfigurationBindingGenerator Condition="'$(EnableConfigurationBindingGenerator)' == ''">true</EnableConfigurationBindingGenerator>
   </PropertyGroup>
 
-  <!-- Enable the interceptors compiler feature by default for projects that use the RequestDelegateGenerator or the ConfigurationBindingGenerator. -->
-  <PropertyGroup Condition="'$(EnableRequestDelegateGenerator)' == 'true' Or '$(EnableConfigurationBindingGenerator)' == 'true'">
-    <Features>$(Features);InterceptorsPreview</Features>
+  <!-- Set the namespaces emitted by the RequestDelegateGenerator for interception when applicable. -->
+  <PropertyGroup Condition="'$(EnableRequestDelegateGenerator)' == 'true'">
```
Comment (layomia) [inline src/WebSdk/ProjectSystem/Targets/Microsoft.NET.Sdk.Web.ProjectSystem.targets:63]: Did you consider setting the conditions on the properties instead, and using the same group for both?

**Review by layomia — COMMENTED** [review]
Looks good.

**Review by layomia — APPROVED** [review]

**Review by eerhardt — APPROVED** [review]

**Comment by captainsafia** [comment]
Merging. This is not a breaking change and doesn't introduce new behavior.

**Pull request merged**